Content.Server/Traits/Assorted/IlliterateSystem.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Content.Server.Arcade;
+using Content.Server.Communications;
+using Content.Shared.Traits.Assorted;
+using Robust.Shared.Random;
+using static Content.Shared.Paper.SharedPaperComponent;
+
+namespace Content.Server.Traits.Assorted;
+
+public sealed class IlliterateSystem : SharedIlliterateSystem
+{
+
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<WriteAttemptEvent>(OnWriteAttempt);
+        SubscribeLocalEvent<ReadAttemptEvent>(OnReadAttempt);
+        SubscribeLocalEvent<CommunicationConsoleAnnouncementEvent>(OnAnnouncement);
+    }
```

Comment:
There's no reason for illiteratesystem to be subscribing to its own event, you should just have callers call illiteratesystem directly instead as it's unnecessary overhead.

Reply:
Removed, callers use the method directly